code/obj/storage/vis_storage_controller.dm

```diff
@@ -0,0 +1,22 @@
+/datum/vis_storage_controller
+	var/obj/parent = null
+	///The items this storage controller contains
+	var/list/obj/vis_items = list()
+
+	proc/initialize(new_parent)
+		parent = new_parent
+	proc/add_item(var/obj/O)
+		vis_items += O
+		O.AddComponent(/datum/component/storage_viscontents, container = src)
+
+	proc/hide()
+		for (var/obj/O in vis_items)
+			parent.vis_contents -= O
+
+	proc/show()
+		for (var/obj/O in vis_items)
+			parent.vis_contents += O
```

Comment:
If `vis_items` is always only going to contain objects, then `as anything` should be used within the `for` loops. This skips an implicit typecheck performed by BYOND, saving processing time.

```
for (var/obj/O as anything in vis_items)
```